mwclient/client.py

```diff
@@ -969,7 +969,21 @@ def ask(self, query, title=None):
         API doc: https://semantic-mediawiki.org/wiki/Ask_API
 
         Returns:
-            Generator for retrieving all search results
+            Generator for retrieving all search results.
+            Iterating over the generator gives a dictionnary containing the answers
+            to the query.
+            The dictionnary is empty if either:
+               - the query is valid but there are no answers,
+               - or if the query is not valid.
+
+        Examples:
+
+            >>> query = "[[Category:my cat]]|[[Has name::a name]]|?Has property"
+            >>> answer = site.ask(query)
+            >>> for a in answer:
+            >>>     for key,value in a.iteritems() # or a.items() in python 3
```

Comment:
Btw. `a.items()` works in both Python 2 and 3, it just doesn't work as an iterator in Python 2, so for the sake of this example, I think it would be better to use `a.items()` and skip the comment.